### PR TITLE
support mmpose2.0 for sdk

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

support mmpose2.0 for sdk

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)
None
## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by irexyc** [comment]
I think the precision difference may due to dump info of postprocess.

The contents of `test_cfg` between master and dev-1.x of mmpose is different. 
In dev-1.x, it moves some contents of `test_cfg` to `codec`, you can compare the two configs list below:

https://github.com/open-mmlab/mmpose/blob/master/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/mspn50_coco_256x192.py#L66-L70

https://github.com/open-mmlab/mmpose/blob/dev-1.x/configs/body_2d_keypoint/topdown_heatmap/coco/td-hm_mspn50_8xb32-210e_coco-256x192.py#L35-L41

File: mmdeploy/codebase/mmpose/deploy/pose_detection.py [hunk 1]
```diff
@@ -311,15 +310,17 @@ def get_preprocess(self, *args, **kwargs) -> Dict:
     def get_postprocess(self, *args, **kwargs) -> Dict:
         """Get the postprocess information for SDK."""
         component = 'UNKNOWN'
+        params = copy.deepcopy(self.model_cfg.model.test_cfg)
```
Comment (irexyc) [inline mmdeploy/codebase/mmpose/deploy/pose_detection.py:313]: I think a more general way is to read the content of model_cfg.model.head.decoder, or if we modify the content of mspn codec like modify the `post_process` to `default`, the converted pipeline.json still have megvii instead.

**Review by irexyc — APPROVED** [review]

File: mmdeploy/codebase/mmpose/deploy/pose_detection.py (hunk 1 above)
Comment (RunningLeon) [reply mmdeploy/codebase/mmpose/deploy/pose_detection.py:313]: post_process would be updated by line 314-321

**Review by AllentDan — APPROVED** [review]
LGTM

**Comment by RunningLeon** [comment]
TODO may update exporting to sdk based on codecs type
```
    def get_postprocess(self, *args, **kwargs) -> Dict:
        """Get the postprocess information for SDK."""
        codec = self.model_cfg.codec
        if isinstance(codec, (list, tuple)):
            codec = codec[0]
        component = 'UNKNOWN'
        params = copy.deepcopy(self.model_cfg.model.test_cfg)
        if self.model_cfg.model.type == 'TopdownPoseEstimator':
            if codec.type == 'MSRAHeatmap':
                params['post_process'] = 'default'
                component = 'TopdownHeatmapSimpleHeadDecode'
            elif codec.type == 'UDPHeatmap':
                pass
            elif codec.type == 'MegviiHeatmap':
                params['post_process'] = 'megvii'
                params['modulate_kernel'] = self.model_cfg.kernel_sizes[-1]
                component = 'TopdownHeatmapMSMUHeadDecode'
            elif codec.type == 'SimCCLabel':
                pass
            elif codec.type == 'RegressionLabel':
                pass
            else:
                raise RuntimeError(f'Unsupported codecs type: {codec.type}')
```

**Pull request merged**